externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala

```diff
@@ -139,6 +146,58 @@ class ExecuteStatement(
     setState(OperationState.FINISHED)
   }
 
+  private def runSetOperation(setOperation: SetOperation): Unit = {
+    if (setOperation.getKey.isPresent) {
+      val key: String = setOperation.getKey.get.trim
+
+      if (setOperation.getValue.isPresent) {
+        val newValue: String = setOperation.getValue.get.trim
+        executor.setSessionProperty(sessionId, key, newValue)
+      }
+
+      val value = executor.getSessionConfigMap(sessionId).getOrDefault(key, "")
+      resultSet = ResultSet.builder
+        .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
+        .columns(Column.physical("set", DataTypes.STRING()))
+        .data(Array(Row.of(PROPERTY_FORMAT.format(key, value))))
+        .build
+    } else {
+      // show all properties if set without key
+      val properties: util.Map[String, String] = executor.getSessionConfigMap(sessionId)
+
+      val entries = ArrayBuffer.empty[Row]
+      properties.forEach((key, value) => entries.append(Row.of(PROPERTY_FORMAT.format(key, value))))
+
+      if (entries.nonEmpty) {
+        val prettyEntries = entries.sortBy(_.getField(0).asInstanceOf[String])
+        resultSet = ResultSet.builder
+          .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
+          .columns(Column.physical("set", DataTypes.STRING()))
```

Comment:
Is it more convenient to return 2 columns, one for the key and the other for the value?

Reply:
> Is it more convenient to return 2 columns, one for the key and the other for the value?

Yes, I thought about this as well. But I found spark/hive/flink are returning key-value pairs, so I adopt the common format.

Reply:
Sorry, I was executing the set query via our in-house platform, not native spark-sql. Maybe I get the result because the platform is using beeline. I'm okay with two separate columns.

Reply:
Let's return 2 columns then. Actually, for CLIs like beeline, users may not have any opportunity to handle results col-by-col, but for callers of JDBC-API, 2 columns make a big improvement on UX.